contracts/Strategy.sol

```diff
@@ -274,60 +371,317 @@ contract Strategy is BaseStrategy {
+    {
+        bentoBox.transfer(
+            BIERC20(address(want)),
+            address(this),
+            address(kashiPairs[kashiPairIndex].kashiPair),
+            sharesToDeposit
+        );
+
+        uint256 depositedFraction =
+            kashiPairs[kashiPairIndex].kashiPair.addAsset(
+                address(this),
+                true,
+                sharesToDeposit
+            );
+
+        if (kashiPairs[kashiPairIndex].pid != 0) {
+            masterChef.deposit(
+                kashiPairs[kashiPairIndex].pid,
+                depositedFraction
+            );
+        }
+    }
+
+    function liquidateKashiPair(uint256 kashiPairIndex, uint256 sharesToFree)
+        internal
+        returns (uint256 _shareLiquiduated)
+    {
+        (, uint256 lastAccrued, ) =
+            kashiPairs[kashiPairIndex].kashiPair.accrueInfo();
+        if (block.timestamp > lastAccrued) {
+            // We need to call accrue to accurately calculate totalAssets
+            kashiPairs[kashiPairIndex].kashiPair.accrue();
+        }
+
+        uint256 liquidShares = kashiPairLiquidShares(kashiPairIndex);
+        if (sharesToFree > liquidShares) {
+            sharesToFree = liquidShares;
+        }
+
+        uint256 fractionsToFree =
+            bentoSharesToKashiFraction(
+                kashiPairs[kashiPairIndex].kashiPair,
+                sharesToFree,
+                true
             );
 
-            kashiPair.addAsset(address(this), true, sharesInBento);
+        // Remove from masterChef if there is a non-zero pid
+        if (kashiPairs[kashiPairIndex].pid != 0) {
+            uint256 fractionInMc = kashiFactionInMasterChef(kashiPairIndex);
+            uint256 fractionsToFreeFromMc = fractionsToFree;
+            if (fractionsToFreeFromMc > fractionInMc) {
+                fractionsToFreeFromMc = fractionInMc;
+            }
+            masterChef.withdraw(
+                kashiPairs[kashiPairIndex].pid,
+                fractionsToFreeFromMc
+            );
+        }
+
+        uint256 fractionBalance = kashiFractionInPair(kashiPairIndex);
+
+        if (fractionsToFree > fractionBalance) {
+            fractionsToFree = fractionBalance;
+        }
+
+        _shareLiquiduated = kashiPairs[kashiPairIndex].kashiPair.removeAsset(
+            address(this),
+            fractionsToFree
+        );
+
+        // Redeposit into the masterChef if there's some spare change
+        uint256 looseKashiFraction = kashiFractionInPair(kashiPairIndex);
+        if (kashiPairs[kashiPairIndex].pid != 0) {
+            masterChef.deposit(
+                kashiPairs[kashiPairIndex].pid,
+                looseKashiFraction
+            );
         }
     }
 
+    //sell all function
+    function sell() internal {
+        uint256 sushiBal = balanceOfSushi();
+        if (sushiBal == 0) {
+            return;
+        }
+
+        sushiRouter.swapExactTokensForTokens(
+            sushiBal,
+            uint256(0),
+            path,
```

Comment:
why using a custom path instead of a SUSHI -> WETH -> WANT?

Reply:
Good question, for many of the stable coins that we would provide, sushi->stable is the lowest slippage route. Nonetheless, i should probably make the route via weth the default.